src/mjolnir/ferry_connections.cc

```diff
@@ -380,35 +380,46 @@ void ReclassifyFerryConnections(const std::string& ways_file,
           if (ret2.second) {
             outbound_path_found = true;
           }
+
+          // Reclassify the first/start edge if a connection to higher class roads
+          // is found. Do this AFTER finding shortest path so we do not immediately
+          // determine we hit the specified classification
+          sequence<Edge>::iterator element = edges[edge.second];
+          auto update_edge = *element;
+          if (ret1.second && ret2.second && update_edge.attributes.importance > kFerryUpClass) {
+            update_edge.attributes.importance = kFerryUpClass;
+            update_edge.attributes.reclass_ferry = remove_destonly;
+            element = update_edge;
+            total_count++;
+          }
         } else {
           // Check if oneway inbound to the ferry
           bool inbound =
               (edge.first.sourcenode_ == node_itr.position()) ? edge_rev_access : edge_fwd_access;
           auto ret = ShortestPath(node_itr.position(), end_node_idx, ways, way_nodes, edges, nodes,
                                   inbound, remove_destonly);
           total_count += ret.first;
-          if (ret.second > 0) {
+          if (ret.second) {
             if (inbound) {
               inbound_path_found = true;
             } else {
               outbound_path_found = true;
             }
           }
-        }
-        ferry_endpoint_count++;
 
-        // Reclassify the first/start edge if a connection to higher class roads
-        // is found. Do this AFTER finding shortest path so we do not immediately
-        // determine we hit the specified classification
-        sequence<Edge>::iterator element = edges[edge.second];
-        auto update_edge = *element;
-        if (inbound_path_found && outbound_path_found &&
-            update_edge.attributes.importance > kFerryUpClass) {
-          update_edge.attributes.importance = kFerryUpClass;
-          update_edge.attributes.reclass_ferry = remove_destonly;
-          element = update_edge;
-          total_count++;
+          // Reclassify the first/start edge if a connection to higher class roads
+          // is found. Do this AFTER finding shortest path so we do not immediately
+          // determine we hit the specified classification
+          sequence<Edge>::iterator element = edges[edge.second];
+          auto update_edge = *element;
+          if (ret.second && update_edge.attributes.importance > kFerryUpClass) {
```

Comment:
nit: the `ret.second` check that you fixed above is already happening in this scope, you could just move this inside there?